sdk/identity/azure_identity/README.md

```diff
@@ -1,77 +1,425 @@
+## Key concepts
+
+### Credentials
+
+A credential is a class that contains or can obtain the data needed for a service client to authenticate requests. Service clients across the Azure SDK accept a credential instance when they're constructed, and use that credential to authenticate requests.
+
+The Azure Identity library focuses on OAuth authentication with Microsoft Entra ID. It offers various credential classes capable of acquiring a Microsoft Entra access token. See the [Credential classes](#credential-classes "Credential classes") section for a list of this library's credential classes.
+
+### DefaultAzureCredential
+
+`DefaultAzureCredential` simplifies authentication while developing apps that deploy to Azure by combining credentials used in Azure hosting environments with credentials used in local development. For more information, see [DefaultAzureCredential overview][dac_overview].
+
+#### Continuation policy
+
+As of version 1.14.0<!-- TODO: I'm not sure what version in the rust libraries this statement becomes true at. My guess would be at our first official beta version. Maybe in that case we can just remove the version specification since this will always be true when using the official library. -->, `DefaultAzureCredential` attempts to authenticate with all developer credentials until one succeeds, regardless of any errors previous developer credentials experienced. For example, a developer credential may attempt to get a token and fail, so `DefaultAzureCredential` will continue to the next credential in the flow. Deployed service credentials stop the flow with a thrown exception if they're able to attempt token retrieval, but don't receive one. Prior to version 1.14.0, developer credentials would similarly stop the authentication flow if token retrieval failed, but this is no longer the case.
+
+This allows for trying all of the developer credentials on your machine while having predictable deployed behavior.
+
+## Examples
+
+The following examples are provided:
+<!-- no toc -->
+* [Authenticate with DefaultAzureCredential](#authenticate-with-defaultazurecredential "Authenticate with DefaultAzureCredential")
+<!-- TODO: Un-Comment if/when the sections are added>
+* [Define a custom authentication flow with ChainedTokenCredential](#define-a-custom-authentication-flow-with-chainedtokencredential "Define a custom authentication flow with ChainedTokenCredential")
+* [Async credentials](#async-credentials "Async credentials")
+</!-->
+
+### Authenticate with `DefaultAzureCredential`
+
+More details on configuring your environment to use `DefaultAzureCredential` can be found in the class's [reference documentation][default_cred_ref].
+
+This example demonstrates authenticating the `CosmosClient` from the [azure_data_cosmos][azure_data_cosmos] library using `DefaultAzureCredential`.
 
 ```rust
-use azure_core::credentials::TokenCredential;
+use azure_data_cosmos::CosmosClient;
```

Comment:
Once we have `SecretClient` ready, we should use that. It's by far the cheapest resource to use if people want to follow along and probably one of the most simple clients.

Reply:
Updated example to use Key Vault Secrets client.